core/src/test/java/io/aiven/klaw/TopicRequestValidatorImplIT.java

```diff
@@ -0,0 +1,276 @@
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.TestPropertySource;
+
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = UiapiApplication.class)
+@TestPropertySource(locations = "classpath:test-application-rdbms.properties")
+@AutoConfigureMockMvc
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DirtiesContext
+public class TopicRequestValidatorImplIT {
+
+  public static final String KWUSER = "kwuser";
+  @Autowired private Validator validator;
+
+  @MockBean private CommonUtilsService commonUtilsService;
+
+  @MockBean private TopicControllerService topicControllerService;
+
+  @MockBean private MailUtils mailService;
+
+  private UtilMethods utilMethods;
+
+  @Mock private UserDetails userDetails;
+
+  @BeforeEach
+  public void mockDefaults() {
+    utilMethods = new UtilMethods();
+    loginMock();
+  }
+
+  @Test
+  @Order(1)
+  public void isValidTestNotAuthorizedUser() {
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(true);
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString()).contains(ApiResultStatus.NOT_AUTHORIZED.value);
+  }
+
+  @Test
+  @Order(2)
+  public void isValidTestTenantFiltering() {
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn(KWUSER);
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("2"));
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.toString())
+        .contains("Failure. Not authorized to request topic for this environment.");
+  }
+
+  @Test
+  @Order(3)
+  public void isValidTestTopicName() {
+    TopicRequestModel addTopicRequest = utilMethods.getTopicRequestModel(1001);
+    when(commonUtilsService.isNotAuthorizedUser(any(), any())).thenReturn(false);
+    when(topicControllerService.getUserName()).thenReturn(KWUSER);
+    when(topicControllerService.getEnvsFromUserId(any())).thenReturn(List.of("1"));
+
+    addTopicRequest.setTopicname("");
+    Set<ConstraintViolation<TopicRequestModel>> violations = validator.validate(addTopicRequest);
+    assertThat(violations.size()).isEqualTo(2);
+    assertThat(violations.toString()).contains("Failure. Please fill in a valid topic name.");
+    assertThat(violations.toString()).contains("Invalid topic name");
```

Comment:
```suggestion
    assertThat(violations.toString()).contains("Failure. Please fill in a valid topic name.", "Invalid topic name");
```
assertj allows multiple contains

Reply:
Didn't know this. nice one. updated.